builtins/src/cost_modeling.rs

```diff
@@ -0,0 +1,162 @@
+//! Configurations for handling cost modeling of builtin programs.
+
+use {
+    crate::BUILTINS,
+    ahash::AHashMap,
+    lazy_static::lazy_static,
+    solana_feature_set::FeatureSet,
+    solana_pubkey::Pubkey,
+    solana_sdk_ids::{ed25519_program, secp256k1_program},
+};
+
+/// Configuration for cost modeling of a builtin program.
+#[derive(Debug)]
+pub struct CostModelingConfig {
+    /// The default cost of the builtin program.
+    /// If None, this builtin program is not cost modeled.
+    pub default_cost: Option<u64>,
+    /// Configuration for updating the cost of the builtin program.
+    /// If None, the cost is never updated.
+    pub new_cost_config: Option<NewCostModelingConfig>,
+}
+
+/// Configuration for updating the cost of a builtin program.
+#[derive(Clone, Copy, Debug)]
+pub struct NewCostModelingConfig {
+    /// The new cost of the builtin program.
+    pub new_cost: u64,
+    /// The feature gate to trigger the new cost.
+    pub feature_id: Pubkey,
+}
+
+struct BuiltinCost {
+    default_cost: Option<u64>,
+    core_bpf_migration_feature: Option<Pubkey>,
+    new_cost_config: Option<NewCostModelingConfig>,
+}
+
+lazy_static! {
+    static ref BUILTIN_INSTRUCTION_COSTS: AHashMap<Pubkey, BuiltinCost> = BUILTINS
+        .iter()
+        .map(|builtin| {
```

Comment:
maybe can `filter` out those `default_cost.is_none()` so the `BUILTIN_INSTRUCTION_COSTS` itself is identical to before-version?

Reply:
I didn't filter out `default_cost.is_none()` only because this allows `get_builtin_instruction_cost` to query the `new_default_cost` field based on feature set.

Reply:
I believe my refactor does this now!